core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -162,6 +162,7 @@ class Partition(val topic: String,
    */
   def makeLeader(controllerId: Int,
                  partitionStateInfo: PartitionStateInfo, correlationId: Int): Boolean = {
+    var leaderHWIncremented = false
```

Comment:
I'm a bit confused with the terminology. How is producing a value a side-effect? That is contradictory as far as I am concerned. From where I stand, returning a value is quite satisfying when compared to mutating a variable. It is also the idiomatic way to write the code in Scala. But maybe I'm missing something.